### PR TITLE
Fix Misleading Error Message on unreadable Media due to Permission

### LINKED ISSUE #1576: Misleading error message when file is not readable
### Description

When a music file is found but cannot be read due to directory permissions the error message is misleading. It also appears in the UI as a 'dummy' entry with no duration/metadata and the file name as a title.

The workardound is simple - fix the permissions.

### Expected Behaviour

I would expect the error to be clear in the logs and (maybe) for the file to not be registered in the UI.

### Steps to reproduce

1. Copy a file to a directory scanned and readable by navidrome 
2. Make sure that the user running navidrome can't read it, e.g. `chown root:root FILE.mp3; chmod 600 FILE.mp3`
3. Execute a quick scan

### Platform information

 - Navidrome version: v0.47.0
 - Browser and version: Firefox 96.0.1
 - Operating System: openSUSE Leap 15.3/Docker image/Kubernetes

### Additional information

Any other information that may be relevant or give context to the problem.

 - Logs?

```
time="2022-01-21T11:44:31Z" level=warning msg="Error reading metadata from file. Skipping" error="cannot process /music/Timpuri Noi/Basca abundentei/04 Adeline.mp3" filePath="/music/Timpuri Noi/Basca abundentei/04 Adeline.mp3"
```

### PR DESCRIPTION
**Pull request by reopt999**
This PR is to fix misleading error message when Navidrome tries to extract the metadata from a media file that is unreadable due to Navidrome process has no permission to read the media file. Additionally, this PR is to fix Navidrome from storing dummy entry from media that are unreadable. Error message may be changed and some checks are probably needed to make it more consistent.

Tested on Archlinux.

Closes #1576

### CODE REVIEW

**Comment by reopt999** [comment]
Hi, @deluan there is a failed check, I'm not sure what is wrong. I noticed that other PR got annotations on which code caused the failure but I don't see any for this commit. Is there anything I can do?

**Review by deluan — CHANGES_REQUESTED** [review]
Thanks for this. The failed check is not your fault, it is caused by new formatting rules in Go 1.18. Just rebase your PR and it should work.

Can you please try to come up with a test for this? Maybe create a test file (copying from `tests/fixtures/test.mp3` and changing its permissions in the test setup?

File: scanner/metadata/taglib/taglib.go [hunk 1]
```diff
@@ -13,7 +13,10 @@ type parsedTags = map[string][]string
 func (e *Parser) Parse(paths ...string) (map[string]parsedTags, error) {
 	fileTags := map[string]parsedTags{}
 	for _, path := range paths {
-		fileTags[path] = e.extractMetadata(path)
+		tags := e.extractMetadata(path)
+		if tags != nil {
+			fileTags[path] = tags
```
[the remark this answers is not in the document]
Comment (reopt999) [reply scanner/metadata/taglib/taglib.go:26]: Thank you for the review, I have pushed a change to classify the error and only the unreadable file should be skipped

**Comment by deluan** [comment]
Hey, I fix the tests (the assertions were not quite right) and did some clean-up as well. I still have to test this on Windows before merging, but in the meantime let me know if you have any concerns with my changes

**Review by deluan — APPROVED** [review]
LGTM. Still pending testing on Windows, hopefully will have it tested this weekend. Thanks!

**Pull request merged**